src/util/Constants.js

```diff
@@ -43,6 +46,7 @@ exports.DefaultOptions = {
   messageCacheMaxSize: 200,
   messageCacheLifetime: 0,
   messageSweepInterval: 0,
+  messageEditHistoryMaxSize: 100,
```

Comment:
Honestly... This imo can be lower - like 50, otherwise lgtm

Reply:
I chose 100 to help with ensuring this wouldn't be a breaking change for 99% of use-cases. Considering we're going from Infinity to this by default, I think it'd be wise to stick with a larger number for now, and it can be reduced in the future if folks this it's appropriate?

Reply:
introducing any sort of default here will make this major, no matter how low or high the number is

Reply:
Ah, fair enough. Is it best to default this to -1 one then, or would it be better to set an aggressive default and keep this as a major change?

Reply:
can surely introduce it with no limit in minor to give people access to use it if they choose to do so by explicitly providing the option (and thus being aware of the change) and change that to be more aggressive for v13 in a separate PR.

Reply:
Makes sense to me, will set this to -1.